app/controllers/collection_numbers_controller.rb

```diff
@@ -186,20 +193,17 @@ def show_selected_collection_numbers(query, args = {})
 
   def build_collection_number
     normalize_parameters
-    if @collection_number.name.blank?
-      flash_error(:create_collection_number_missing_name.t)
+    if @collection_number.name.blank? || @collection_number.number.blank?
+      flash_error(missing_attribute_msg.t)
```

Comment:
I'd write this something like this:
```
return if missing_attribute
if name_and_number_used?
  ...
```
and then have
```
def missing attribute
  error_message = missing_attribute_error_message
  return false unless error_message
  flash_error(error_message.t)
  return true
end

def missing_attribute_error_message
  return :missing_attribute_error_message if @collection_number.name.blank?
  return :create_collection_number_missing_number if @collection_number.number.blank?
end
```
I haven't tested the above, but I think it's a bit DRYer and more Ruby-ish.

Reply:
Done in Commit #b7de553